main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2827,9 +2827,45 @@ object Classpaths {
 
   val jvmPublishSettings: Seq[Setting[_]] = Seq(
     artifacts := artifactDefs(defaultArtifactTasks).value,
-    packagedArtifacts := packaged(defaultArtifactTasks).value
+    packagedArtifacts := {
+      if (sbtPlugin.value) packagedSbtPluginArtifacts.value
+      else packaged(defaultArtifactTasks).value
```

Comment:
Ok, I applied the suggestion.

I think the `Def.ifS` is not necessary because the `Def.task` macro is smart enough to handle that. But maybe it makes things slightly clearer.

Reply:
@adpi2 just for info, we found that the `if` macro does not work for SBT 1.2.x (and possibly some versions above that). Not sure if it makes a difference.